src/librustc_mir/transform/inline.rs

```diff
@@ -616,8 +616,9 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Integrator<'a, 'tcx> {
                 Operand::Consume(Lvalue::Local(l)) => *local = l,
                 ref op => bug!("Arg operand `{:?}` is {:?}, not local", idx, op)
             }
+        } else {
+            *local = self.local_map[Local::new(idx - self.args.len())];
         }
-        *local = self.local_map[Local::new(idx - self.args.len())];
```

Comment:
Is this required to get the miri test suite to pass?

Reply:
Yes. We test inlined code in the test suite.

Reply:
But the test suite currently passes even without this change...?

Reply:
Because the change isn't in a nightly yet